sched/signal/sig_nanosleep.c

```diff
@@ -261,6 +262,11 @@ int clock_nanosleep(clockid_t clockid, int flags,
 {
   int ret;
 
+  if (clockid < CLOCK_REALTIME || clockid > CLOCK_BOOTTIME)
+    {
+      return EINVAL;
+    }
```

Comment:
@xiaoxiang781216 @patacongo do you know what should have more "power" here, cancelation point or invalid parameter check? In other words, should these lines be placed here or after `enter_cancellation_point();`? Or in other words, should `clock_nanosleep()` call with invalid `clockid` be equivalent to `pthread_testcancel` or not?

Reply:
> @xiaoxiang781216 @patacongo do you know what should have more "power" here, cancelation point or invalid parameter check? In other words, should these lines be placed here or after `enter_cancellation_point();`? Or in other words, should `clock_nanosleep()` call with invalid `clockid` be equivalent to `pthread_testcancel` or not?

enter_cancellation_point() catches the case where the thread is canceled prior to calling the cancellation point function.  I would think that returning ECANCEL would be more important to return in that case.  Otherwise the thread may not exit until the next time a cancellation point function is called.